File: src/java/Factorial.java
import java.math.BigInteger;

public class Factorial {
    public static void main(String[] args) {
        String countEnv = System.getenv("COUNT");
        int count = 200;
        if (countEnv != null) {
            try {
                count = Integer.parseInt(countEnv);
            } catch (NumberFormatException e) {
                // Keep default
            }
        }

        // System.out.println("Calculating Factorial(" + count + ")...");
        
        BigInteger factorial = BigInteger.ONE;

        long start = System.nanoTime();
        for (int i = 1; i <= count; i++) {
            factorial = factorial.multiply(BigInteger.valueOf(i));
        }
        long end = System.nanoTime();
        double durationMs = (end - start) / 1_000_000.0;

        System.out.println("Result(" + count + "!): " + factorial);
        System.out.printf("Time: %.3f ms%n", durationMs);
    }
}

File: src/js/factorial.js
const count = parseInt(process.env.COUNT) || 200;

// console.log(`Calculating Factorial(${count})...`);

let factorial = 1n; // Use BigInt

for (let i = 1; i <= count; i++) {
    factorial *= BigInt(i);
}

console.log(`Result(${count}!): ${factorial.toString()}`);
